src/mongocxx/gridfs/bucket.cpp

```diff
@@ -254,48 +316,102 @@ downloader bucket::_open_download_stream(const client_session* session,
     options::find chunks_options;
     chunks_options.sort(chunks_sort.extract());
 
+    if (start and end) {
+        if (*start > *end) {
+            throw gridfs_exception{error_code::k_invalid_parameter,
+                                "expected end to be greater than start"};
+        }
+    }
+
+    if (start and *start > 0) {
+        if (*start > file_len) {
+            throw gridfs_exception{error_code::k_invalid_parameter,
+                        "expected start to not be greater than the file length"};
+        }
+        auto start_offset_div = std::lldiv(*start, chunk_size);
+        start_offset.chunks_offset = start_offset_div.quot;
+        start_offset.bytes_offset = start_offset_div.rem;
+        chunks_options.skip(start_offset.chunks_offset);
+    }
+
+    if (end) {
+        if (*end > file_len) {
+            throw gridfs_exception{error_code::k_invalid_parameter,
+                        "expected end to not be greater than the file length"};
+        }
+        if (*end < file_len) {
+            const int32_t batch_size = 1 + ((*end - *start) / chunk_size);
+            chunks_options.batch_size(batch_size);
```

Comment:
```suggestion
            const int32_t num_chunks = 1 + ((*end - *start) / chunk_size);
            chunks_options.limit(num_chunks);
```

IIUC the intention is to limit the number of total chunks returned.

`batch_size` does not limit the number of documents returned on a cursor.

See https://docs.mongodb.com/manual/reference/command/find/ and https://docs.mongodb.com/manual/reference/command/getMore/ for a description of the server handling of `batchSize` and `limit`.

Reply:
Done in commit e956be47938a8f5913baa52fec46d9bbc2efb01f.